File: VB/E4121VB/Controllers/DbGeneratorController.vb
Imports System.Web.Mvc

Namespace Controllers
    Public Class DbGeneratorController
        Inherits Controller

        Public Function Create() As ActionResult
            Return New JsonResult With {
                .Data = DatabaseGenerator.TryCreateDatabase(GetTableKey())
            }
        End Function

        Public Function GetRecordCount() As ActionResult
            Return New JsonResult With {
                .Data = DatabaseGenerator.GetCreatingDatabaseRecordCount(GetTableKey())
            }
        End Function

        Private Function GetTableKey() As String
            Return Request.Params("tableKey")
        End Function

        Protected Overrides Sub OnException(ByVal filterContext As ExceptionContext)
            filterContext.Result = New ContentResult() With {
                .Content = filterContext.Exception.Message,
                .ContentType = "text/plain"
            }
            filterContext.ExceptionHandled = True
            Response.StatusCode = 500
        End Sub
    End Class
End Namespace

File: CS/E4121/Controllers/DbGeneratorController.cs
using E4121.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace E4121.Controllers
{
    public class DbGeneratorController : Controller
    {
        public ActionResult Create()
        {
            return new JsonResult
            {
                Data = DatabaseGenerator.TryCreateDatabase(GetTableKey())
            };
        }
        public ActionResult GetRecordCount()
        {
            return new JsonResult
            {
                Data = DatabaseGenerator.GetCreatingDatabaseRecordCount(GetTableKey())
            };
        }

        string GetTableKey()
        {
            return Request.Params["tableKey"];
        }

        protected override void OnException(ExceptionContext filterContext)
        {
            filterContext.Result = new ContentResult()
            {
                Content = filterContext.Exception.Message,
                ContentType = "text/plain"
            };
            filterContext.ExceptionHandled = true;
            Response.StatusCode = 500;
        }
    }
}